lib/rofl_ofdpa_fm_driver.cpp

```diff
@@ -158,10 +160,18 @@ cofflowmod rofl_ofdpa_fm_driver::enable_port_pvid_ingress(uint8_t ofp_version,
   fm.set_match().set_in_port(port_no);
   fm.set_match().set_vlan_vid(0);
 
+  if (vrf_id) {
+    fm.set_instructions()
+        .set_inst_apply_actions()
+        .set_actions()
+        .add_action_set_field(i)
+        .set_oxm(ofdpa::coxmatch_ofb_vrf(vrf_id));
+  }
+
   fm.set_instructions()
       .set_inst_apply_actions()
       .set_actions()
-      .add_action_set_field(cindex(0))
+      .add_action_set_field(++i)
```

Comment:
That would evaluate into `.add_action_set_field(cindex(1))` regardless wether vrf_id is set or not.
Suggest to post-increment i in the new function above.